pkg/volume/util/atomic_writer.go

```diff
@@ -193,6 +195,14 @@ func (w *AtomicWriter) Write(payload map[string]FileProjection) error {
 	}
 
 	// (8)
+	if setPerms != nil {
+		if err := setPerms(); err != nil {
```

Comment:
As a reader I am confused - if we don't pass the name of the dir we just made, what is `setPerms()` supposed to do?  Maybe we should pass subPath to the function and then the volumes can choose to disregard it if they want and do the whole volume instead (and they can comment why).

Can you add a note to the function comment explaining the contract with setPerms (and that it is optional).

Reply:
I've added the parameter and comment to volume implementations. I called it `tsDir` instead of `subPath` because while updating the test case I realized the path inside `Write()` was full path and not just the subdir.